rust/src/rfb/rfb.rs

```diff
@@ -633,3 +633,86 @@ pub unsafe extern "C" fn rs_rfb_register_parser() {
         SCLogDebug!("Protocol detector and parser disabled for RFB.");
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_error_state() {
+        let mut state = RFBState::new();
+
+        let buf = &[
+            0x05, 0x00, 0x03, 0x20, 0x20, 0x18, 0x00, 0x01, 0x00, 0xff, 0x00, 0xff, 0x00, 0xff,
+            0x10, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x1e, 0x61, 0x6e, 0x65, 0x61,
+            0x67, 0x6c, 0x65, 0x73, 0x40, 0x6c, 0x6f, 0x63, 0x61, 0x6c, 0x68, 0x6f, 0x73, 0x74,
+            0x2e, 0x6c, 0x6f, 0x63, 0x61, 0x6c, 0x64, 0x6f, 0x6d, 0x61, 0x69, 0x6e,
+        ];
+        let r = state.parse_request(buf);
+
+        assert_eq!(
+            r,
+            AppLayerResult {
+                status: -1,
+                consumed: 0,
+                needed: 0
+            }
+        );
+    }
+    #[test]
+    fn test_request_response_events() {
```

Comment:
Adding some feedback after discussing this with Victor :)
So, he sees this as testing the state machine for the RFB app-proto, and that's a positive thing, and this is the right place for it. We just need some adjustments, to make it easier to understand ;)
1) Change the test name to something like `test_state_machine`
2) Add a comment explaining what's the test purpose - since it's a more extensive one and its workings are not immediately clear